In [1]:
import pandas as pd
import numpy as np
import scipy.stats as stats

In [6]:
def one_sample_t_test(data,assumed_mean):
    return stats.ttest_1samp(a=data,popmean=assumed_mean)

def two_sample_t_test(data1,data2):
    result = stats.ttest_ind(a=data1,b=data2,equal_var=True)
    return result

In [20]:
data = pd.read_csv('data_1.csv')

In [24]:
raj_data = data[data['State name']=='RAJASTHAN']
ass_pop_mean = 2050000
print(one_sample_t_test(raj_data['Population'],ass_pop_mean))

Ttest_1sampResult(statistic=0.13908506369155546, pvalue=0.8902545833414418)


One Sample T-test on Rajasthan Population:

Hypothesis: The mean of Rajasthan's population is 2050000                                                                       
Inference: From the test, it is observed that the p-value is 0.89 which is greater than the alpha value 0.05. So the hypothesis            is condidered to be true i.e., the average of the state's population is around 2050000

In [62]:
and_data = data[data['State name']=='ANDHRA PRADESH']
tam_data = data[data['State name']=='TAMIL NADU']
print(two_sample_t_test(and_data['Agricultural_Workers'],tam_data['Agricultural_Workers']))

Ttest_indResult(statistic=7.3682922866177005, pvalue=1.1523122250650383e-09)


Two Sample T-test for comparing Andhra Pradesh and Tamil Nadu Agriculture Workers:

Hypothesis: The no. of agriculture workers in Andhra Pradesh and Tamil Nadu are nearly equal.                                   
Inference: Fom the test, it is observed that the p-value is 1.1523122250650383e-09 i.e., it is equal to 1.15 × 10^-9 which is very less than the threshold 0.05. So the hypothesis is not true. And the positive test statistic 7.36 indicates that Andhra Pradesh has more agriculture workers than Tamil Nadu. 

In [60]:
guj_data = data[data['State name']=='GUJARAT']
raj_data = data[data['State name']=='RAJASTHAN']
print(two_sample_t_test(raj_data['Households_with_Television'],guj_data['Households_with_Television']))

Ttest_indResult(statistic=-2.125151040532665, pvalue=0.0379211137725814)


Two Sample T-test for comparing Households with TV's in Gujarath and Rajasthan:

Hypothesis: Gujarat and Rajasthan have almost same no. of households with Televisions.                                   
Inference: Fom the test, it is observed that the p-value is 0.03 and hence the hypothesis taken is false. And the negative test statistic -2.12 indicates that Gujarath has less households with TV's than Rajasthan. 

In [63]:
man_data = data[data['State name']=='MANIPUR']
meg_data = data[data['State name']=='MEGHALAYA']
print(two_sample_t_test(man_data['Urban_Households'],meg_data['Urban_Households']))

Ttest_indResult(statistic=-0.02493617681681884, pvalue=0.9804578201402745)


Two Sample T-test for comparing Urban Households in Manipur and Meghalaya:

Hypothesis: Urban houses are nearly same in Manipur and Meghalaya.                                   
Inference: Fom the test, it is observed that the p-value is 0.98 and hence the hypothesis is considered to be true strongly i.e., Manipur and Meghalaya have nearly same number of urban households. 